files/grest/rpc/01_cached_tables/epoch_info_cache.sql

```diff
@@ -155,6 +168,17 @@ BEGIN
       LEFT JOIN epoch_param ep ON ep.epoch_no = e.no
       LEFT JOIN cost_model cm ON cm.id = ep.cost_model_id
       INNER JOIN block b ON b.time = e.start_time
+      LEFT JOIN LATERAL (
+        SELECT
+          e.no,
+          SUM(r.amount) as amount
+        FROM
+          reward r
+        WHERE
+          r.earned_epoch = e.no
+        GROUP BY
+          e.no
+      ) reward_pot ON TRUE
     WHERE
       e.no >= _epoch_no_to_insert_from
```

Comment:
Joins in general aren't executed in any particular order as they are ordered in code. But... lateral joins always execute last as they loop over all results from previous joins. Where filtering is then applied to the final result, this is at least my understanding of it. 

To reduce rows lateral join has to loop over, you could move the where filter to the inner join on block. This would force it to filter out these results before lateral join is applied. I don't know if the query planner is smart enough to figure this out by itself or what the actual gain might be. But could be worth testing. 

So inner join would then be:
`INNER JOIN block b ON b.time = e.start_time AND e.no >= _epoch_no_to_insert_from`
and WHERE filter can be removed.

Reply:
Thanks, good point. I will give it a try